lib/tile.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Tile do
+  defstruct [:type, :state, :adjacent_bomb_count]
+
+  def new(:empty), do: new_with_type(:empty)
+  def new(:bomb), do: new_with_type(:bomb)
+
+  defp new_with_type(type) do
+    %Tile{
```

Comment:
I like the idea of these structs coming in to play, I think they really help with making the code easier to grok. This feels a bit object-oriented to me, but I don't really know much about polymorphism in functional languages. I can't wait to read your blog post! 😄